lib/cog/models/relay.ex

```diff
@@ -24,11 +24,24 @@ defmodule Cog.Models.Relay do
     model
     |> Repo.preload(:groups)
     |> cast(params, @required_fields, @optional_fields)
+    |> allow_user_defined_id_on_insert(params)
     |> validate_presence_on_insert(:token)
     |> encode_token
     |> unique_constraint(:name)
   end
 
+  def allow_user_defined_id_on_insert(changeset, params) do
+    case {get_field(changeset, :id), Map.get(params, "id")} do
```

Comment:
@vanstee will `get_change(changeset, :id)` return nil if we are doing an update? I don't think the id is included in changes on update. If that's the case I think we would get a "cannot modify ID" failure. 

Would `get_field(changeset, :id)` work better?